test/unit/pages/location/controller-test.js

```diff
@@ -30,4 +50,48 @@ describe('LocationController', function() {
     });
 
   });
+
+  describe('folderChanged', function() {
```

Comment:
You can remove this method, I'll refactor to have only checkFolder() method in controller.